tests/drivers/hwinfo/reset_cause/src/main.c

```diff
@@ -0,0 +1,357 @@
+{
+	ARG_UNUSED(wdt_dev);
+	ARG_UNUSED(channel_id);
+	wdt_status = WDT_HAS_FIRED;
+}
+
+/* Print LOG delimiter. */
+static void print_bar(void)
+{
+	LOG_INF("===================================================================");
+}
+
+static void print_supported_reset_cause(void)
+{
+	int32_t ret;
+
+	/* Store supported reset causes in global variable placed at NOINIT_SECTION. */
+	ret = hwinfo_get_supported_reset_cause((uint32_t *) &supported);
+
+	/* Print which reset causes are supported. */
+	if (ret == 0) {
+		LOG_INF("Supported reset causes are:");
+		if (supported & RESET_PIN) {
+			LOG_INF(" 0: RESET_PIN is supported");
+		} else {
+			LOG_INF(" 0: no support for RESET_PIN");
+		}
+		if (supported & RESET_SOFTWARE) {
+			LOG_INF(" 1: RESET_SOFTWARE is supported");
+		} else {
+			LOG_INF(" 1: no support for RESET_SOFTWARE");
+		}
+		if (supported & RESET_BROWNOUT) {
+			LOG_INF(" 2: RESET_BROWNOUT is supported");
+		} else {
+			LOG_INF(" 2: no support for RESET_BROWNOUT");
+		}
+		if (supported & RESET_POR) {
+			LOG_INF(" 3: RESET_POR is supported");
+		} else {
+			LOG_INF(" 3: no support for RESET_POR");
+		}
+		if (supported & RESET_WATCHDOG) {
+			LOG_INF(" 4: RESET_WATCHDOG is supported");
+		} else {
+			LOG_INF(" 4: no support for RESET_WATCHDOG");
+		}
+		if (supported & RESET_DEBUG) {
+			LOG_INF(" 5: RESET_DEBUG is supported");
+		} else {
+			LOG_INF(" 5: no support for RESET_DEBUG");
+		}
+		if (supported & RESET_SECURITY) {
+			LOG_INF(" 6: RESET_SECURITY is supported");
+		} else {
+			LOG_INF(" 6: no support for RESET_SECURITY");
+		}
+		if (supported & RESET_LOW_POWER_WAKE) {
+			LOG_INF(" 7: RESET_LOW_POWER_WAKE is supported");
+		} else {
+			LOG_INF(" 7: no support for RESET_LOW_POWER_WAKE");
+		}
+		if (supported & RESET_CPU_LOCKUP) {
+			LOG_INF(" 8: RESET_CPU_LOCKUP is supported");
+		} else {
+			LOG_INF(" 8: no support for RESET_CPU_LOCKUP");
+		}
+		if (supported & RESET_PARITY) {
+			LOG_INF(" 9: RESET_PARITY is supported");
+		} else {
+			LOG_INF(" 9: no support for RESET_PARITY");
+		}
+		if (supported & RESET_PLL) {
+			LOG_INF("10: RESET_PLL is supported");
+		} else {
+			LOG_INF("10: no support for RESET_PLL");
+		}
+		if (supported & RESET_CLOCK) {
+			LOG_INF("11: RESET_CLOCK is supported");
+		} else {
+			LOG_INF("11: no support for RESET_CLOCK");
+		}
+		if (supported & RESET_HARDWARE) {
+			LOG_INF("12: RESET_HARDWARE is supported");
+		} else {
+			LOG_INF("12: no support for RESET_HARDWARE");
+		}
+		if (supported & RESET_USER) {
+			LOG_INF("13: RESET_USER is supported");
+		} else {
+			LOG_INF("13: no support for RESET_USER");
+		}
+		if (supported & RESET_TEMPERATURE) {
+			LOG_INF("14: RESET_TEMPERATURE is supported");
+		} else {
+			LOG_INF("14: no support for RESET_TEMPERATURE");
+		}
+	} else if (ret == -ENOSYS) {
+		LOG_INF("hwinfo_get_supported_reset_cause() is NOT supported");
+		supported = 0;
+	} else {
+		LOG_ERR("hwinfo_get_supported_reset_cause() failed (ret = %d)", ret);
```

Comment:
Maybe it would be easier for varication if in case of error there is assert instead of log?